1/auth

xx3u opened the pull request:
- Updated views for RegisterView (register form for new users);
- Added functional tests for view;
- Updated model for Profile extending User django model;
- Added ProfielInLine in UserAdmin;
- Updated travis.yml for encrypted file;
- Updated pytest for 100 coverage and installed flake8;

xx3u committed:
Updated functional test for view
Extended User model using one-to-one link in Profile

xx3u committed:
Updated test for register view
Updated flake8

xx3u committed:
Update encrypted file for travis

xx3u committed:
Update pytest for flake8

xx3u committed:
Update Readme for build pass and codcoverage

xx3u committed:
Updated .gitignore for .xml

Commit:
Merge pull request #2 from xx3u/4/Readme

4/readme

khazidhea commented on website/test/functional/test_views.py:
```diff
@@ -0,0 +1,58 @@
+    response = client.post(
+        '/login/', {'username': 'test', 'password': 'test'}
+    )
+    assert response.status_code == 302
```
Add follow redirects and check for response code 200

xx3u replied:
`redirect = HttpResponse(response)`
`assert redirect.status_code == 200`

khazidhea commented on website/test/functional/test_views.py:
```diff
@@ -0,0 +1,58 @@
+        '/login/', {'username': 'test', 'password': 'wrongpassword'}
+    )
+    assert response.status_code == 200
+    assert b'Please enter a correct username and password.' in response.content
```
Compare str vs response.text

xx3u replied:
response.text not identified, so instead of that decoding is used:
`assert 'Please enter a correct username and password.' in response.content.decode()`

khazidhea commented on website/models.py:
```diff
@@ -5,12 +5,13 @@
     )
 
 
 class Profile(models.Model):
```
Use AbstractUser

xx3u replied:
AbstractUser added into models.py and update in other files

xx3u committed:
Update tests for comments on github:
update test_login for redirect and change response status code from 302 to 200;
update test login fail for response.content.decode();

xx3u committed:
Add CustomUser based on AbstractUser

khazidhea commented on website/models.py:
```diff
@@ -1,17 +1,16 @@
+
+class CustomUser(AbstractUser):
+    birthday = models.DateField(null=True, blank=True)
+    contact_number = models.CharField(max_length=20, null=True)
```
CharField ставится blank=True, практически никогда не ставится null=True

xx3u replied:
Updated for blank=True instead of null=True

khazidhea commented on website/models.py:
```diff
@@ -1,17 +1,16 @@
+from django.contrib.auth.models import AbstractUser
+
+
+class CustomUser(AbstractUser):
```
Назови просто User, писать везде CustomUser неудобно

xx3u replied:
Updated in 5/profile for Profile class instead of CustomUser

khazidhea commented on website/test/functional/test_views.py:
```diff
@@ -1,20 +1,24 @@
     )
-    assert response.status_code == 302
+    redirect = HttpResponse(response)
+    assert login is True
```
Лучше поставь этот assert сразу после логина

xx3u replied:
changed place of assert

khazidhea commented on website/test/functional/test_views.py:
```diff
@@ -1,20 +1,24 @@
-    assert response.status_code == 302
+    redirect = HttpResponse(response)
+    assert login is True
+    assert redirect.status_code == 200
```
Нет, это некрасиво и непонятно. Как говорил добавь follow=True у client.post

xx3u replied:
updated tests for follow=True instead of redirect

xx3u committed:
Update in models contact_number as blank=True instead of null=True;
Update test login for status code==200 with follow=True;

khazidhea commented on website/test/functional/test_views.py:
```diff
@@ -0,0 +1,60 @@
+    response = client.post(
+        '/register/', {'username': 'test', 'password': 'test'}
+    )
+    assert response.status_code == 200
```
Проверить создание объекта User
